File: 0019 remove-nth-node-from-end-of-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int i=0;
        ListNode dummy = new ListNode(0,head);
        ListNode temp = dummy;
        ListNode prev=dummy;
        while(temp!=null) {
            if(i>n) prev=prev.next;
            temp=temp.next;
            i++;
        }
        prev.next=prev.next.next;
        return dummy.next;
    }
}

File: 0019 remove-nth-node-from-end-of-list/solution.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(0,head);
        ListNode curr = dummy;
        ListNode prev = curr;
        int i=0;
        while(curr!=null) {
            if(i>n) prev=prev.next;
            curr=curr.next;
            i++;
        }
        prev.next = prev.next.next;
        return dummy.next;
    }
}
